### DISCUSSION TITLE
Include Pico SDK?

### DISCUSSION THREAD

**Discussion by oerkel47** [earlephilhower/arduino-pico#189] [Q&A] [answered]
Hi and first of all thank you for this project.

I would like to use some functions of the original API like accessing registers and getting some more control to maybe port some libraries. How would I go about including the SDK to my project? 

I can't even manage to find the place where the arduino functions are mapped to the SDK. Thats how it works right?

Thanks.

**Comment by earlephilhower** [comment]
It's already in there.  Just `#include <...>` whatever file you want and use the SDK calls.  You can look at the `cores/rp2040` directory or the `SPI` or `Wire` libraries to see how it works.

**Reply by oerkel47** [reply]
Yep I know it's in there..I just don't get how the folder structure works with these includes. For example `#include <hardware/spi.h>` or `#include <hardware/pio.h>`..Like where are these files actually? There are multiple files and folders with more or less same names. I am a bit confused.

**Comment by earlephilhower** [comment, marked as answer]
It's part of the install.  The repo has a "pico-sdk" directory which is just a git submodule of the official Pico-SDK.  Your code doesn't need to worry about it, the core automatically adds the necessary include paths.  Other than any conflicts between your calls and the core's infrastructure, I don't see any big issue.

If you want to really understand what's going on, you should probably go with `git clone` (see readme) vs. the IDE GUI install, simply because you control things 100% in that case.  But either way, most Pico-SDK should "just work."

**Reply by oerkel47** [reply]
I see. Thanks a lot for clearing that up.
